File: valorant-coach-frontend/src/types/dataset.ts
export interface DatasetListItem {
  id: string;
  name: string;
  source: string;
  player_name: string | null;
  agent: string | null;
  map_name: string | null;
  filename: string;
  file_size_bytes: number | null;
  duration_seconds: number | null;
  analysis_id: string | null;
  status: string;
  analysis_progress: number | null;
  analysis_status_text: string | null;
  created_at: string | null;
}

export interface DatasetResponse {
  id: string;
  name: string;
  description: string | null;
  source: string;
  player_name: string | null;
  team: string | null;
  agent: string | null;
  map_name: string | null;
  rank: string | null;
  tags: string[] | null;
  filename: string;
  file_size_bytes: number | null;
  duration_seconds: number | null;
  analysis_id: string | null;
  status: string;
  analysis_progress: number | null;
  analysis_status_text: string | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface DatasetAnalysisStatus {
  id: string;
  status: string;
  progress: number;
  status_text: string | null;
  analysis_id: string | null;
  knowledge_entries_count: number | null;
}

export interface DatasetStats {
  total_datasets: number;
  pro_datasets: number;
  user_datasets: number;
  total_size_bytes: number;
  total_duration_seconds: number;
  agents: Record<string, number>;
  maps: Record<string, number>;
}

export interface DatasetUploadPayload {
  file: File;
  name: string;
  description?: string;
  source?: string;
  player_name?: string;
  team?: string;
  agent?: string;
  map_name?: string;
  rank?: string;
  tags?: string;
}

File: valorant-coach-backend/app/schemas/dataset.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class DatasetCreate(BaseModel):
    name: str
    description: Optional[str] = None
    source: str = "user"  # "user" or "pro"
    player_name: Optional[str] = None
    team: Optional[str] = None
    agent: Optional[str] = None
    map_name: Optional[str] = None
    rank: Optional[str] = None
    tags: Optional[list[str]] = None


class DatasetUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    source: Optional[str] = None
    player_name: Optional[str] = None
    team: Optional[str] = None
    agent: Optional[str] = None
    map_name: Optional[str] = None
    rank: Optional[str] = None
    tags: Optional[list[str]] = None


class DatasetResponse(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    source: str
    player_name: Optional[str] = None
    team: Optional[str] = None
    agent: Optional[str] = None
    map_name: Optional[str] = None
    rank: Optional[str] = None
    tags: Optional[list[str]] = None
    filename: str
    file_size_bytes: Optional[int] = None
    duration_seconds: Optional[float] = None
    analysis_id: Optional[str] = None
    status: str
    analysis_progress: Optional[int] = None
    analysis_status_text: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class DatasetListItem(BaseModel):
    id: str
    name: str
    source: str
    player_name: Optional[str] = None
    agent: Optional[str] = None
    map_name: Optional[str] = None
    filename: str
    file_size_bytes: Optional[int] = None
    duration_seconds: Optional[float] = None
    analysis_id: Optional[str] = None
    status: str
    analysis_progress: Optional[int] = None
    analysis_status_text: Optional[str] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class DatasetUploadResponse(BaseModel):
    id: str
    message: str
